tests/commands/run/test_integration.py

```diff
@@ -1,21 +1,135 @@
-import tempfile, shutil
+import yaml, pytest
 from ...util import *
+from ...fixtures import *
 from .util import *
 from sinol_make import configure_parsers
 
-def test_run_simple():
-	with tempfile.TemporaryDirectory() as tmpdir:
-		package_path = os.path.join(tmpdir, "abc")
-		shutil.copytree(get_simple_package_path(), package_path)
+def test_simple(create_package):
+	"""
+	Test a simple run.
+	"""
+	package_path = create_package
+	command = get_command()
+	create_ins_outs(package_path, command)
 
-		command = get_command()
-		create_ins(package_path, command)
-		os.chdir(package_path)
-		create_outs(package_path, command)
-		os.chdir(package_path)
+	parser = configure_parsers()
 
-		parser = configure_parsers()
+	args = parser.parse_args(["run"])
+	command = Command()
+	command.run(args)
 
-		args = parser.parse_args(["run"])
-		command = Command()
+
+def test_no_expected_scores(capsys, create_package):
+	"""
+	Test with no sinol_expected_scores in config.yml.
+	Should run, but exit with exit code 1.
```

Comment:
In the code you're checking more than the exit code. We also can add what in the output you're checking.

Adding good comment messages greatly helps in understanding the code (even though they are tests). Testing codes tend to have a lot of IO and exceptions etc. which slows down reading the code.

Reply:
I added better comments: 32873ce9a2f8f9c627efa0c1dc127260b0c08569